packages/react-router/src/path.ts

```diff
@@ -241,7 +243,36 @@ export function matchPathname(
 }
 
 export function removeBasepath(basepath: string, pathname: string) {
-  return basepath != '/' ? pathname.replace(basepath, '') : pathname
+  switch (true) {
```

Comment:
wouldn't it be much simpler to replace using

```tsx
new RegExp(`^${basepath}`)
```

notice the `^` to ensure it only replaces the basepath if it occurs at the start of the string

Reply:
1) your solution does not ensure full-segment match `/app` -> `/application` will still occur.
2) such interpolation with no escaping (and there is no built-in regex-ascaping in JS AFAIK) is prone to errors, as URI's path syntax allows symbols that has meaning within regexes `/[some-segment]` obviously different outcome than expected.
3) i have no proof fo that but my gut feeling tells compillation of regex for each invocation will be suboptimal compared to bunch of `if`s

Reply:
1. true, did not think about that
2. hmm I'd say not super likely, but possible
3. I wasn't optimizing for runtime performance, just wanted to keep the code complexity down if possible

thanks for the detailed response